src/main/java/bookshop/dto/BookDto.java

```diff
@@ -1,14 +1,24 @@
 package bookshop.dto;
 
+import jakarta.validation.constraints.NotBlank;
+import jakarta.validation.constraints.NotNull;
+import jakarta.validation.constraints.Positive;
 import java.math.BigDecimal;
 import lombok.Data;
 
 @Data
 public class BookDto {
+    @NotNull
+    @Positive
     private Long id;
+    @NotBlank
     private String title;
+    @NotBlank
     private String author;
+    @NotBlank
     private String isbn;
```

Comment:
ISBN could be also validated for a format correspondence. You can use a built-in annotation for it.

Reply:
done